### PR TITLE
Rename Allowlisted and Denylisted to Allowed and Denied for plugin loading

### PR DESCRIPTION
**Pull request by bobisme**
### Background
I noticed an inclusive-language change where terms like "white" and "black" were replaced with "allow" and "deny" which resulted in awkward non-words like "allowlisted."  Simpler terms would make the code cleaner and clearer.

### Changes

* Changed "allowlist" and "denylist" to thinks like "allowed" and "denied" for more natural language.
* Note: This affects the ALLOWLISTED_PLUGINS env variable.
* Also noticed the config for denied plugins was unimplemented, so I added that.

### Documentation

Functions were already documented.

### Test Plan
Unit tests already covered this functionality.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/config/config.py [hunk 1]
```diff
@@ -133,12 +133,13 @@ def __init__(self) -> None:
...
-        if plugins_allowlist:
-            self.plugins_allowlist = plugins_allowlist.split(",")
-        else:
-            self.plugins_allowlist = []
-        self.plugins_denylist = []
+        self.plugins_allowed = os.getenv("ALLOWED_PLUGINS", [])
+        if self.plugins_allowed:
+            self.plugins_allowed = self.plugins_allowed.split(",")
+
+        self.plugins_denied = os.getenv("DENIED_PLUGINS")
```
Comment (lc0rp) [inline autogpt/config/config.py]: Nice. I was about to implement this.

A slight suggestion to ensure self.plugins_denied is always a list, even when unset, matching the ALLOWED_PLUGINS behavior.

```
self.plugins_denied = os.getenv("DENIED_PLUGINS", [])
```

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

File: .env.template [hunk 2]
```diff
@@ -248,13 +248,13 @@ OPENAI_API_KEY=your-openai-api-key
...
 ################################################################################
 
-#ALLOWLISTED_PLUGINS - Sets the listed plugins that are allowed (Example: plugin1,plugin2,plugin3)
-#DENYLISTED_PLUGINS - Sets the listed plugins that are not allowed (Example: plugin1,plugin2,plugin3)
-ALLOWLISTED_PLUGINS=
-DENYLISTED_PLUGINS=
+#ALLOWED_PLUGINS - Sets the listed plugins that are allowed (Example: plugin1,plugin2,plugin3)
+#DENIED_PLUGINS - Sets the listed plugins that are not allowed (Example: plugin1,plugin2,plugin3)
+ALLOWED_PLUGINS=
+DENIED_PLUGINS=
```
Comment (Pwuts) [inline .env.template:257]: This makes sense 👍🏼

File: autogpt/config/config.py [hunk 3]
```diff
@@ -159,17 +159,17 @@ def __init__(self) -> None:
         self.plugins: List[AutoGPTPluginTemplate] = []
         self.plugins_openai = []
 
-        plugins_allowlist = os.getenv("ALLOWLISTED_PLUGINS")
-        if plugins_allowlist:
-            self.plugins_allowlist = plugins_allowlist.split(",")
+        plugins_allowed = os.getenv("ALLOWED_PLUGINS")
```
Comment (Pwuts) [inline autogpt/config/config.py:162]: `plugins_allowed` is a bit of an ambiguous name: is it a `bool` indicating whether plugins are allowed, or a `list` of allowed plugins? `plugins_allowlist` made perfect sense here imo

File: autogpt/plugins.py [hunk 4]
```diff
@@ -251,8 +251,8 @@ def scan_plugins(cfg: Config, debug: bool = False) -> List[AutoGPTPluginTemplate
     return loaded_plugins
 
 
-def denylist_allowlist_check(plugin_name: str, cfg: Config) -> bool:
-    """Check if the plugin is in the allowlist or denylist.
+def check_allowed(plugin_name: str, cfg: Config) -> bool:
```
Comment (Pwuts) [inline autogpt/plugins.py:254]:
```suggestion
def is_allowed(plugin_name: str, cfg: Config) -> bool:
```

**Review by Pwuts — CHANGES_REQUESTED** [review]
I'm thinking we need backwards compatibility for the old keys in `.env`

**Comment by Torantulino** [comment]
@CodiumAI-Agent

**Pull request closed without merging**